### DISCUSSION TITLE
Controller for Array Fields

### DISCUSSION THREAD

**Discussion by the-avid-engineer** [react-hook-form/react-hook-form#9683] [Ideas]
I'm aware that a Controller already "works" with array fields, but it is missing something that would be really nice to have.

Imagine the idea (a so-called ArrayController) is nearly identical to the normal Controller, but the render prop method has the object you get by calling `useFieldArray`, probably named `fieldArray`, in addition to the `field`, `fieldState`, and `formState` objects you normally get as parameters.

Then, you don't have to call useFieldArray directly, and all of the methods/fields on the useFieldArray object are readily available when the ArrayController is mounted/not available when it is not mounted.

Thoughts?

**Comment by leapful** [comment]
@the-avid-engineer 

Thank you for the idea. Since react-hook-form main goal is to reduce the bundle size as much as possible, it would be great to implement the customer controller separately, may be something like https://github.com/react-hook-form/error-message as a util library.
